### DISCUSSION TITLE
<Link> not working when using _app.js instead of index.js

### DISCUSSION THREAD

**Discussion by david-Khayutin** [vercel/next.js#14192] [Help] [answered]
Hey All

Hoping someone can point me into the right direction for this issue. I am using next to build a web application. Since i am using some custom css, I created an _app.js file over the regular index.js file. As a result the link routing does not seem to be working . I tested the same routes when using index.js as my home page and the routes worked. Is there a way to work around this without switching back to index.js?

My file structure is the way its suppose to be. All pages in /pages dir. 

thanks for any help!!!

**Comment by jamesmosier** [comment, marked as answer]
Hi @david-Khayutin! `_app.js` is not a replacement for `index.js`. `_app.js` is a helper file that is useful for global CSS, maintaining state, persisting layout, etc. Each `page` is rendered inside of `_app` via `<Component {...pageProps} />` 

* `pages/index.js` would be accessible at `http://localhost:3000`
* `page/_app.js` is not accessible at any route, instead each page uses it as a "wrapper"

[See the docs on `_app`](https://nextjs.org/docs/advanced-features/custom-app)

**Reply by david-Khayutin** [reply]
hey awesome.. I misunderstood how that worked. I fixed it up and it works great!!

thanks @jamesmosier
